helix-core/src/increment/date_time.rs

```diff
@@ -334,16 +334,19 @@ fn add_months(date_time: NaiveDateTime, amount: i64) -> Option<NaiveDateTime> {
 
     let day = cmp::min(date_time.day(), ndays_in_month(year, month));
 
-    Some(NaiveDate::from_ymd(year, month, day).and_time(date_time.time()))
+    NaiveDate::from_ymd_opt(year, month, day).map(|date| date.and_time(date_time.time()))
 }
 
 fn add_years(date_time: NaiveDateTime, amount: i64) -> Option<NaiveDateTime> {
     let year = i32::try_from((date_time.year() as i64).checked_add(amount)?).ok()?;
     let ndays = ndays_in_month(year, date_time.month());
 
     if date_time.day() > ndays {
-        let d = NaiveDate::from_ymd(year, date_time.month(), ndays);
-        Some(d.succ().and_time(date_time.time()))
+        if let Some(d) = NaiveDate::from_ymd_opt(year, date_time.month(), ndays) {
+            d.succ_opt().map(|date| date.and_time(date_time.time()))
+        } else {
+            None
+        }
```

Comment:
```suggestion
        NaiveDate::from_ymd_opt(year, date_time.month(), ndays)
            .and_then(|date| date.succ_opt())
            .map(|date| date.and_time(date_time.time()))
```

Reply:
Added in [this commit](https://github.com/helix-editor/helix/pull/4738/commits/3d623fdb093e048ddd42d143569625ccc3a90b5f).